### DISCUSSION TITLE
Image resizing in fabric.js

### DISCUSSION THREAD

**Discussion by Aech0** [fabricjs/fabric.js#10152] [Q&A] [answered]
is there any way i can make the edges of the image stretchable such that the entire image does not get resized when the image is stretched from one corner. I want the user to adjust the image such that when the middle of the edge of the image is clicked and dragged , the size should increase (horizontally or vertically, depending on the edge) which is currently happening. But when the corner is clicked and dragged, currently the entire size of the image is increasing proportionately , i want that when a corner is clicked and dragged, the image should be stretched from the corner only(like distortion), whilst the other 3 corners are fixed to their previous positions. ultimately, i want a poster to fit in a banner which is tilted.

I tried
 let image = new fabric.Image(imageElement, { left: 50, top: 50, scaleX: 0.5, scaleY: 0.5, skewX: -8, angle: -4, lockUniScaling: false, // Allow non-proportional scaling lockScalingFlip: false, // Allow flipping during scaling scaleUniformly: false, // Ensure independent scaling for width and height cornerSize: 10, // Size of control corners cornerStyle: "circle", // Style of corner controls transparentCorners: false, // Make the corners visible for better control hasControls: true, // Show the corner controls for scaling });

But none of it seems to be working

example is below
<img width="829" alt="Screenshot 2024-09-12 180028" src="https://github.com/user-attachments/assets/d5c74613-8294-4132-8373-3ba25f2c6ebb">

**Comment by asturur** [comment]
you want it to deform? and perspective correct?

**Reply by Aech0** [reply]
yes, it should basically fit the hoarding in the canvas(bg video of the canvas)
any method works for me to implement this functionality, i even tried applying css but that also is not possible I THINK

**Reply by asturur** [reply, marked as answer]
You need the 4 point image transform tool. is not something you can always do with skewing and scaling

**Reply by Aech0** [reply]
got a js code on the web for the same 4 point transform tool, now integrating it with my fabric canvas in reactapp.
Thankyou for the help thou
